### PR TITLE
[mobx-apollo] add definition for mobx-apollo

### PR DESCRIPTION
**Pull request by pselden**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by pselden** [comment]
I couldn't find any docs about this -- how do I actually do tests for things that have dependencies on other libraries?

My test is failing because of: Cannot find module 'apollo-client'. -- when I tested locally I just installed those and assumed there was some magic that would get those installed when running the test on Travis, but obviously that is not the case.

File: types/mobx-apollo/package.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "private": true,
+    "dependencies": {
+        "apollo-client": "^2.0.0",
+        "graphql-tag": "^2.0.0",
```
Comment (pselden) [inline types/mobx-apollo/package.json]: This is actually just a dev dependency, but the linter did not allow that -- are these actually published as the pacakge depdencies?

**Comment by pselden** [comment]
`In /home/travis/build/DefinitelyTyped/DefinitelyTyped/types/mobx-apollo/package.json: Dependency apollo-client not in whitelist; please make a pull request to types-publisher adding it.`

I'm not quite sure what to do here... does every package that has a dependency have to add to this whitelist? The whitelist looks pretty small so that can't be right.

**Comment by pselden** [comment]
Added package to whitelist, now it's passing!

**Comment by pselden** [comment]
@uniqueiniquity  -- did not pass

**Comment by uniqueiniquity** [comment]
Odd, there are no other `package.json` files on DT anywhere afaik. Let me see if I can figure out why this is happening.

**Comment by pselden** [comment]
Removed that commit.

**Comment by pselden** [comment]
There are a few, if you use github search for package.json you'll see maybe 20 or so. Admittedly that is much smaller than I expected there would be -- it seems like most things here depend on types that are published to DefinitelyTyped so the external dependencies are not required. I too would like to figure out the correct way of doing it though. There is not much in the docs about it (I will submit a PR to improving it once we figure out the correct approach).

**Comment by uniqueiniquity** [comment]
I see now; this is the first time I've seen a package that depends on types bundled with the library. You're doing the right thing. Merging now; sorry for the confusion!

**Pull request merged**